### PR TITLE
First restructuring of devel docs

### PR DESCRIPTION
**Pull request by giovannipizzi**
This is preliminary to #3035 to move some docs out
of the main documentation or at least revised.

Moreover, also the main devel section has been
renamed from "Development" to "For AiiDA developers"
and the only page under "AiiDA design" has been moved 
inside the same section "for AiiDA developers".

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -77,12 +76,11 @@ The software is available at http://www.aiida.net.
...
-    :caption: Development
+    :caption: For AiiDA developers
     :hidden:
 
     developer_guide/index
 
-
 .. toctree::
     :maxdepth: 1
     :caption: Tutorials
```
Comment (sphuber) [inline docs/source/index.rst]: Should we also include the sections of `Tutorials` in this restructuring? Moving their content to `StructureData` and `UpfData` respectively? And the `plugin development` sectino should be moved to the temporary `aiida-plugins`, as I think it should be moved out of the `aiida-core` docs

Comment (giovannipizzi) [reply docs/source/index.rst]: I suggest that I mark this in the respective issues, so at least this restructuring is done and we can work on this in the next days during the coding days.
In particular moving tutorials into the section requires probably some thought.

Comment (giovannipizzi) [reply docs/source/index.rst]: For the `plugin development` section, is there something that I have missed, apart for the section that I moved below under the title `aiida-plugins [Section to be moved]`?

Comment (giovannipizzi) [reply docs/source/index.rst]: Added comments for #3019 and #3022 (for the part on `Tutorials`)

**Pull request merged**